config.d.ts

```diff
@@ -16,10 +16,26 @@ declare module '*.yaml' {
     path: string;
   }
 
+  interface HomeSection {
+    title: string;
+    text: string;
+    image: string;
+    imageAlt: string;
+    buttonText?: string;
+    buttonRoute?: string;
+  }
+
+  interface ContactUs {
+    text: string;
+  }
+
```

Comment:
More context

Reply:
```tsx
<Section sectionConfig={} even={} key={}/>
```
```tsx
let section: JSX.Element | undefined; // get rid of undefiend case by using an external mapping function
switch (sectionConfig.sectionType) [
  case "textSection":
    section = <TextSection />
    break;
}
if (section == undefined) throw new Error("Impossible"); 
return (
  <section classname={even ? style.even : style.odd}>
    <SectionHeader />
    {section}
  </section>
)
```
```tsx
getSectionElement();
```
```tsx
sectionConfig={}
```

Reply:
Hey Jake, can we throw this on the backburner? I understand your approach but this seems like slightly too much config driven development, especially for right now where it might be better to just get everything done and then go back to it later

Reply:
Changed